src/libstd/rt/unwind/gcc.rs

```diff
@@ -98,6 +99,20 @@ pub mod eabi {
                                 context: *mut uw::_Unwind_Context)
             -> uw::_Unwind_Reason_Code;
     }
+    #[cfg(target_os = "nacl")]
+    unsafe fn __gcc_personality_v0(_version: c_int,
+                                   _actions: uw::_Unwind_Action,
+                                   _exception_class: uw::_Unwind_Exception_Class,
+                                   _ue_header: *mut uw::_Unwind_Exception,
+                                   _context: *mut uw::_Unwind_Context) ->
+        uw::_Unwind_Reason_Code
+    {
+        // we just need to be here to prevent linker errors.
+        // based on my analysis of PNaClSJLJ.cpp and ExceptionInfoWriter.cpp,
+        // the personality functions are just discarded without any consultation.
+        use core::intrinsics::abort;
+        abort()
+    }
```

Comment:
Newlib doesn't provide it, no.

It doesn't translate to an abort-on-panic. PNaCl doesn't have intrinsic support for EH: EH support is provided through the `-pnacl-sjlj-eh` pass, which, through my fork and, soon, Rust proper, `rustc` uses to rewrite the LLVM-level EH constructs into SJLJ EH, and the EH `landingpad` info into a static table (the personality is thrown away; in fact even `rust_eh_personality*` is unneeded). From there, `libcxx` reads into that static table as it bubbles up the stack (ie it uses it's own personality function).

`-pnacl-sjlj-eh` is toolchain defined and heavily tied to a specific personality. In fact, due to PNaCl platform independance most of the `_Unwind_*` functions are unavailable, making `__gcc_personality_v0` useless anyway.

I should also note that my fork targeting PNaCl passes all but 59 of the `run-pass` tests and all but 6 of the `run-fail` tests. The tests take several hours, so I save the log [here](https://gist.github.com/DiamondLovesYou/df4df48c1bccf13c7dd8) if you'd like to look.